doc/md/knowledge-center/mysql/generated_columns.md

```diff
@@ -0,0 +1,218 @@
+---
+id: mysql-generated-columns
+title: How to Use Generated Columns in MySQL with Atlas
+slug: /knowledge/mysql/generated-columns
+---
+
+**MySQL** is a popular open-source relational database. **Generated columns** are a feature of MySQL that allows you to define tables with columns whose value is a function of the value stored in other columns; without requiring complex expressions in SELECT, INSERT or UPDATE queries.
+
+## What are Generated Columns?
+
+Generated columns are columns that contain values calculated by expressions which can be dependent on other columns; in a similar manner to formulas in a spreadsheet. There are two types of generated columns in MySQL: _Stored_ and _Virtual_.
+
+### Stored Generated Columns
+
+Stored generated columns are stored and evaluated when a row is inserted or updated. As a result, stored generated columns use disk space in addition to CPU cycles during the execution of `INSERT` and `UPDATE` statements.
+
+### Virtual Generated Columns
+
+Virtual generated columns are not stored, and only evaluated when a row is read _(after BEFORE [triggers](https://dev.mysql.com/doc/refman/5.7/en/trigger-syntax.html))_. As a result, virtual generated columns take no storage at the cost of CPU cycles for `SELECT` statements.
```

Comment:
Maybe mention the pros/cons of each.

Reply:
In addition to the use-case ("When to Use Generated Columns"), these are the pros and cons.